### PR TITLE
Add test confirming negative rules

### PR DESCRIPTION
**Pull request by avangelillo**
### What does this PR do?
Add a test to confirm that deny rules are working as intended.  Also a small change to how the function names are determined to make it so that createFunctions can be called multiple times within a single test.

### Motivation
Confirming functionality around allow:false rules.

### Testing Guidelines
* Test only change

### CODE REVIEW

File: integration-tests/lambda-management-events.test.js [hunk 1]
```diff
@@ -72,6 +72,51 @@ describe("Remote instrumenter lambda management event tests", () => {
...
+    // After some time
+    const isInstrumented = await pollUntilTrue(60000, 5000, () =>
+      isFunctionInstrumented(instrumentedFunctionName),
+    );
+    expect(isInstrumented).toStrictEqual(true);
+
+    const isUninstrumented = await isFunctionUninstrumented(
+      notInstrumentedFunctionName,
+    );
+    expect(isUninstrumented).toStrictEqual(true);
```
Comment (clifordshelton) [inline integration-tests/lambda-management-events.test.js:121]: For the case of checking to see if it's instrumented, I assume we're polling because it could take a little bit of time for the lambda instrumentation operation to happen. In that case, I'm worried checking for the negative scenario like this could give false positives in case it did get instrumented but with a delay. Could you implement the test this way so that we can be certain?
- Set the rule
- Create two functions that get instrumented
- Now change the rule so that one of the functions shouldn't be instrumented anymore because of an `allow:false` type rule
- Now await to see if that function gets uninstrumented
- After this passes, also confirm that the other function stays instrumented

Comment (clifordshelton) [reply integration-tests/lambda-management-events.test.js:121]: Or do you think we can reliably depend that by the time the first pollUntilTrue operation completes with instrumentation the other one also must be complete? If that's true, then it makes sense and my above comment doesn't apply

Comment (avangelillo) [reply integration-tests/lambda-management-events.test.js:121]: Because we make the lambda that should not be instrumented first then the one that should be, when the test sees that it is instrumented then we know that the instrumenter would have processed both events.  But I think I like instrumenting it, adjusting the rule, then checking that it gets uninstrumented better since it makes it more explicit

**Review by TalUsvyatsky — APPROVED** [review]
